arraycontext/impl/pytato/utils.py

```diff
@@ -124,4 +125,61 @@ def get_loopy_target(self) -> "lp.PyOpenCLTarget":
 
 # }}}
 
+# {{{ Transfer mappers
+
+
+class TransferToDeviceMapper(CopyMapper):
+    def __init__(self, actx: ArrayContext) -> None:
+        super().__init__()
+        self.actx = actx
+
+    def map_data_wrapper(self, expr: DataWrapper) -> Array:
+        import numpy as np
+
+        if not isinstance(expr.data, np.ndarray):
+            raise ValueError("TransferToDeviceMapper: tried to transfer data that "
+                             "is already on the device")
+
+        new_dw = self.actx.from_numpy(expr.data)
+        assert isinstance(new_dw, DataWrapper)
+        return DataWrapper(
+            data=new_dw.data,
+            shape=expr.shape,
+            axes=expr.axes,
+            tags=expr.tags,
+            non_equality_tags=expr.non_equality_tags)
```

Comment:
Ideally, this should just do `return self.actx.from_numpy(expr.data).tagged(expr.tags)` I think, but there seems to be no way to transfer the `non_equality_tags` in that case.

Reply:
Added as a comment